docs/rules/prefer-array-index-of.md

```diff
@@ -0,0 +1,29 @@
+# Prefer [`Array#indexOf()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/indexOf) over [`Array#findIndex()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/findIndex) when looking for the index of an item
```

Comment:
We normally add links in details. and this should be the same as the on in readme.md